modules/web/src/app/cluster/details/external-cluster/update-external-cluster-machine-deployment-dialog/component.ts

```diff
@@ -137,13 +138,17 @@ export class UpdateExternalClusterMachineDeploymentDialogComponent extends BaseF
     }
   }
 
-  private _getExternalMachineDeploymentPatchModel(): ExternalMachineDeploymentPatch {
+  private _getExternalMachineDeploymentPatchModel(isEKSProvider: boolean): ExternalMachineDeploymentPatch {
+    const endSliceParameter = 4;
     return {
       spec: {
         replicas: this.form.get(Controls.Replicas).value,
         template: {
           versions: {
-            kubelet: this.form.get(Controls.KubeletVersion).value,
+            // in EKS we shouldn't send the patch with the full version, only the major version.
+            kubelet: isEKSProvider
+              ? this.form.get(Controls.KubeletVersion).value.slice(0, endSliceParameter)
```

Comment:
I think we should use the `semver` package for parsing instead of using `slice` just to avoid any possibility of parsing error. For example, if version is `1.23-beta` or `1.100.1` then `slice` would not work as expected.